### PR TITLE
Remove BandBrightness; Wavelength from EmissionLine.

### PR DESCRIPTION
**Pull request by rpiaggio**
The model allowed construction of inconsistent brightness maps in `SpectralDefinition.BandNormalized` and `SpectralDefinition.EmissionLines`: the `Band` of `Wavelength` used as key in each map entry may not coincide with the one inside the `BandBrightness` or `EmissionLine`.

To solve this, `Wavelength` was removed from `EmissionLine`.

The `BandBrightness` class was removed altogether, in favor of adding an `error` value to `Measure`.

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/math/dimensional/measure.scala [hunk 1]
```diff
@@ -12,18 +13,32 @@ import shapeless.tag
...
    */
-  def toQuantity: Quantity[N, units.Type] = value.withUnit[units.Type]
+  def toExactQuantity: Quantity[N, units.Type] = value.withUnit[units.Type]
+
+  /**
+   * Convert error to `coulomb.Quantity`.
+   */
+  def valueQuantity: Option[Quantity[N, units.Type]] = error.map(_.withUnit[units.Type])
+
+  // def withError(error: N): self.type = copy(error = error.some).asInstanceOf[self.type]
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/math/dimensional/measure.scala:28]: Commented code?

File: modules/core/shared/src/main/scala/lucuma/core/model/EmissionLine.scala [hunk 2]
```diff
@@ -10,16 +10,14 @@ import coulomb.cats.implicits._
...
 import eu.timepit.refined.types.numeric.PosBigDecimal
 import lucuma.core.math.BrightnessUnits._
-import lucuma.core.math.Wavelength
 import lucuma.core.math.dimensional._
 import lucuma.core.math.units._
 import monocle.Focus
 import monocle.Lens
 
 final case class EmissionLine[T](
-  wavelength: Wavelength,
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/model/EmissionLine.scala:20]: So this is gone and instead it is in the map's index?

File: modules/tests/shared/src/test/scala/lucuma/core/model/SpectralDefinitionSuite.scala [hunk 3]
```diff
@@ -5,84 +5,76 @@ package lucuma.core.model
...
-
-  val sd1Surface: SpectralDefinition[Surface] = {
-    val b: BandBrightness[Surface] =
-      BandBrightness.withDefaultUnits[Surface](BrightnessValue.fromDouble(10.0), Band.R)
 
+  val sd1Surface: SpectralDefinition[Surface] =
     SpectralDefinition.BandNormalized(
       UnnormalizedSED.StellarLibrary(StellarLibrarySpectrum.A0I),
-      SortedMap(b.band -> b)
+      SortedMap(Band.R -> Band.R.brightnessMeasure[Surface](BrightnessValue.fromDouble(10.0)))
```
Comment (swalker2m) [inline modules/tests/shared/src/test/scala/lucuma/core/model/SpectralDefinitionSuite.scala:50]: Is the `Band.R.brightnessMeasure` just used to pick up the default units for `R` bands?  You could just as well do `Band.R` -> `Band.J.brightnessMeasure[Surface](BrightnessValue.fromDouble(10.0))` in terms of what gets created in the end?

**Review by cquiroz — APPROVED** [review]
Some small comments but LGTM

File: modules/core/shared/src/main/scala/lucuma/core/model/EmissionLine.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/model/EmissionLine.scala:20]: Exactly

File: modules/tests/shared/src/test/scala/lucuma/core/model/SpectralDefinitionSuite.scala (hunk 3 above)
Comment (rpiaggio) [reply modules/tests/shared/src/test/scala/lucuma/core/model/SpectralDefinitionSuite.scala:50]: Yes, it's just used to pick up the default units, and yes, nothing stops us from using the default units from another band.

Maybe we should the remove the `brightnessMeasure` convenience method?

We could still create the brightness with `Band.R.defaultUnits[Surface].withValueTagged(BrightnessValue.fromDouble(10.0))`, where it is more explicit that we are just using the band's default units.

Comment (swalker2m) [reply modules/tests/shared/src/test/scala/lucuma/core/model/SpectralDefinitionSuite.scala:50]: Yes, that sounds good.  I was confused at first when I saw it, assuming that `Band.R` was somehow being associated with the measure being created there.

**Pull request merged**